### PR TITLE
refactor: improve ERC20 utilities and interoperability with Decimal

### PR DESCRIPTION
**Pull request by RedFox20**
Main goal is to make ERC20 utility much easier to use.
Calling `this.toDecimal(await this.contract.getBalance())` allows for a quick conversion to `Decimal`
Also added `Decimal.valueOf()` and `Decimal.equals()` to make integration into unit tests easier.

Refactored ERC20 decimals initialization to have less awaits.

Finally, added new `Addressable` to make it possible to call `token.balanceOf(tempusPool)` and similar functions with any kind of addressable object. Such as ethers.Contract, ethers.SignerWithAddress, string and our ContractBase.

### CODE REVIEW

File: test/utils/ContractBase.ts [hunk 1]
```diff
@@ -7,14 +7,24 @@ import * as abstractSigner from "@ethersproject/abstract-signer/src.ts";
...
-export function addressOf(signer:SignerOrAddress) {
-  if (typeof(signer) === "string")
-    return signer;
-  if (signer.address)
-    return signer.address;
-  throw new Error("Invalid signer (no address): " + signer);
+export function addressOf(addressable: Addressable): string {
+  if (typeof(addressable) === "string")
+    return addressable;
+  if (addressable instanceof ContractBase || addressable instanceof Contract || addressable.address)
```
Comment (axic) [inline test/utils/ContractBase.ts:16]:
```suggestion
  if (addressable instanceof ContractBase || addressable instanceof Contract || addressable instanceof Signer)
```

Comment (RedFox20) [reply test/utils/ContractBase.ts:16]: That's a good change, I added this 👍🏻 
```ts
  if (addressable instanceof ContractBase || addressable instanceof Contract ||
      addressable instanceof signers.SignerWithAddress)
```

File: test/utils/ERC20.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Contract, Transaction } from "ethers";
-import { Numberish } from "./Decimal";
-import { ContractBase, SignerOrAddress, Signer, AbstractSigner, addressOf } from "./ContractBase";
+import { Decimal, Numberish } from "./Decimal";
```
Comment (axic) [inline test/utils/ERC20.ts:2]:
```suggestion
import { Numberish } from "./Decimal";
```

Comment (RedFox20) [reply test/utils/ERC20.ts:2]: No, I left it there intentionally, because I have WIP code which uses it and my next PR will have it anyway

**Review by mijovic — APPROVED** [review]

**Pull request merged**